### PR TITLE
Overhaul tool usage in CI and for contribution

### PR DESCRIPTION
**Pull request by mk-mxp**
- Put all code styling rules into PHPCS config
  - Add XML namespace for completeness
  - Remove non-existent `type="int"`
  - Add directory rules for improved test coverage and required exceptions
  - Fix the new upcoming violations
- PHPCS config renamed to default name
- Moved tool installation to composer
- Moved tool usage to composer
  - Allows running all unit tests now, too
- Overhauled README

This is part of the discussion in [the forum](https://forum.exercism.org/t/php-streamlining-php-codesniffer-usage/8538).

### CODE REVIEW

File: exercises/concept/city-office/.meta/exemplar/Form.php
Comment (neenjaw) [inline exercises/concept/city-office/.meta/exemplar/Form.php:1]: This exercise is intentional without visibility modifiers

File: exercises/concept/windowing-system/.meta/exemplar.php
Comment (neenjaw) [inline exercises/concept/windowing-system/.meta/exemplar.php:1]: This exercise is intentional without visibility modifiers

**Review by neenjaw — CHANGES_REQUESTED** [review]
Couple small changes needed. I'm fine with putting it in one place.

**Comment by mk-mxp** [comment]
Should I do these also?
* Rename the config file to `phpcs.xml`, so it is not necessary to specify it
* Remove `bin/install*.sh` and `bin/lint.sh`, these seem overly complicated and are only used for README instructions. Instead we should point to the track instructions https://exercism.org/docs/tracks/php/installation#h-install-composer and use `composer install`

**Comment by mk-mxp** [comment]
@neenjaw I think, GitHub is confused because I'm not assigned to this PR and so my changes do not count for review resolving.

**Comment by mk-mxp** [comment]
@neenjaw The latest version of PHP CodeSniffer is not compatible to the latest Slevomat Coding Standard. I did exclude the updated PHP CodeSniffer for installation, but that is of course only temporary.

**Comment by mk-mxp** [comment]
Problem:

> "Invalid sniff properties set for individual sniffs will now result in an error and halt the execution of PHPCS"
> PHPCodeSniffer Release notes V3.8.0

Solution:

- [x] Change Slevomat Sniff property to current name

**Comment by neenjaw** [comment]
Can you cherry pick a patch to fix ci to a separate pr to unblock the other prs?

I haven't had time to review this one as it comprises a more complex change for the track.

**Comment by mk-mxp** [comment]
Rebased to solve merge conflicts.

Might it help to split this PR into 2 or 3? One with the rules / changes due to rules, one with the README and contribution workflow changes and maybe the removal of the old script also in one extra?

What I checked for workflow changes / script removal:

- This projects CI does not use the removed scripts. It uses [composer for PHP CodeSniffer](https://github.com/exercism/php/blob/f0f182d0c302a6e20340390a7c75a8ed72d91a5f/.github/workflows/exercise-lint-phpcs-psr-12.yml#L43) and direct [PHAR download for PHPUnit](https://github.com/exercism/php/blob/f0f182d0c302a6e20340390a7c75a8ed72d91a5f/.github/workflows/exercise-tests-phpunit-9.yml#L38)
- [PHP test runner](https://github.com/exercism/php-test-runner/blob/61252274d05daf021fa70c6166edc1b9a0d1ce63/Dockerfile#L12) does it's own thing
- [PHP representer](https://github.com/exercism/php-representer) also has nothing to do with these workflow files
- I haven't found any hint on CI workflows other than `configlet` stuff in Exercism's documentation
- We use `composer` during contribution workflow (locally and in CI) only, with no potential for package version conflicts because of the nature of the exercises
- `composer` is the only tool used for contribution now, running the exact same things in CI and locally (if locally up-to-date, of course)

**Comment by mk-mxp** [comment]
@ErikSchierboom Please advise on the following:

This MR changes the contribution workflow for the PHP track massively. One part is removing `install.sh` and `install-*` completely. They are not used in CI since long and now replaced by `composer install` for contributors.

Does this have any negative impact on you or other parts of exercism (e.g. mass MRs)?

File: README.md [hunk 1]
```diff
@@ -1,53 +1,60 @@
...
-[![Codacy Badge](https://api.codacy.com/project/badge/Grade/68242198cd124a3ebcbdc291d0e0eda4)](https://www.codacy.com/app/borgogelli/php?utm_source=github.com&utm_medium=referral&utm_content=borgogelli/php&utm_campaign=Badge_Grade)
-
 Exercism exercises in PHP
 
-## Install Dependencies
+Follow these instructions, if you want to contribute to the track.
+You must have a BASH shell to run the tooling locally.
 
-### All dependencies
+If you only want to solve the exercises, go to the [track docs][track-docs] and choose your installation option there.
```
Comment (homersimpsons) [inline README.md:8]:
```suggestion
Follow these instructions to contribute to the PHP track.

To solve the execise, head to the [PHP track](https://exercism.org/tracks/php) and check the [documentation][track-docs].
```

File: README.md [hunk 2]
```diff
@@ -1,53 +1,60 @@
...
 
-```shell
-> ./bin/install.sh
-```
+## Install Dependencies
 
-### Only tests dependencies
+You must have installed `composer` as a global tool as recommended in the [PHP track installation docs][composer-installation-docs].
```
Comment (homersimpsons) [inline README.md:12]:
```suggestion
The following system dependencies are required:
- `composer`, as recommended in the [PHP track installation docs][composer-installation-docs].
- [`bash` shell](https://www.gnu.org/software/bash/)
```

File: README.md [hunk 3]
```diff
@@ -1,53 +1,60 @@
...
 
 ```shell
-> ./bin/install-phpunit-9.sh
+bin/fetch-configlet
+composer install
```
Comment (homersimpsons) [inline README.md:16]:
```suggestion
Run the following command to get started with this project:

```shell
bin/fetch-configlet # The official tool for managing Exercism language track repositories
composer install # Required dependencies to develop this track
```

File: README.md [hunk 4]
```diff
@@ -1,53 +1,60 @@
...
 ```
 
-### Only style-check dependencies
+You now have all tools required to contribute.
```
Comment (homersimpsons) [inline README.md:19]:
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -1,53 +1,60 @@
...
-```
+## Running Exercism resources management
+
+`bin/configlet` is a tool to manage exercism resources in this track.
+See [Building Exercism docs][configlet-docs].
 
 ## Running Unit Test Suite
 
-### PHPUnit 9
+We use PHPUnit 9.6.x and a shell loop injecting `exemplar.php` as the solution for testing:
```
Comment (homersimpsons) [inline README.md:28]:
```suggestion
The tests are run with PHPUnit. A shell loop injecting `exemplar.php` is provided to ease testing.

Execute the following command to run the tests:
```

File: README.md [hunk 6]
```diff
@@ -1,53 +1,60 @@
...
 
 ```shell
-> PHPCS_BIN="./bin/phpcs.phar" PHPCS_RULES="./phpcs-php.xml" ./bin/lint.sh
+composer lint:check
+```
+
+To auto-fix the coding styles:
+
+```shell
+composer lint:fix
```
Comment (homersimpsons) [inline README.md:45]:
```suggestion
This project use a slightly [modified] version of [PSR-12], use the following commands to apply code style:
```shell
composer lint:check # Checks the files against the code style rules
composer lint:fix # Automatically fix codestyle issues
```

File: phpcs.xml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  <arg value="n" />
+
+  <!-- Show sniffs (it's easy to find solutions knowing the code) -->
+  <arg value="s" />
+
+  <!-- A TAB is 4 chars wide (does not replace them, for calculation only!) -->
+  <arg name="tab-width" value="4" />
+
+  <!-- Run on 10 files in parallel -->
+  <arg name="parallel" value="10" />
```
Comment (homersimpsons) [inline phpcs.xml:22]: I like to use a value of 60, it does a full line and better scale with a lot of core.
```suggestion
  <!-- Run on 60 files in parallel -->
  <arg name="parallel" value="60" />
```

**Review by homersimpsons — CHANGES_REQUESTED** [review]
Let's get this merged, I did suggest some re-wording, feel free to discuss them.

File: phpcs.xml (hunk 7 above)
Comment (mk-mxp) [reply phpcs.xml:22]: I didn't see any difference on my computer, 1.5 seconds anyways. But that's no value I care for...

**Review by homersimpsons — APPROVED** [review]

**Pull request merged**